standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/txn/TxnHandler.java

```diff
@@ -2725,7 +2725,7 @@ private void insertTxnComponents(long txnid, LockRequest rqst, Connection dbConn
           }
           String dbName = normalizeCase(lc.getDbname());
           String tblName = normalizeCase(lc.getTablename());
-          String partName = normalizeCase(lc.getPartitionname());
+          String partName = lc.getPartitionname();
```

Comment:
@nareshpr , do you know if partition name part (<name>=value) is already normalized here?

Reply:
Yes, i validated below 4 SQL's with my patch, partition(key=value) is already normalized 

insert into table abc PARTITION(CitY='Bangalore') values('Dan');
insert overwrite table abc partition(CiTy='Bangalore') select Name from abc;
update table abc set Name='xy' where CiTy='Bangalore';
delete from abc where CiTy='Bangalore';

Reply:
Could there be an issue if HMS API is used outside of Hive, for example via Impala?

Reply:
I doubt HMS API's will hit this flow. Let me know any example or testcase. If required, i can change code to split key=value & convert key to lowerCase and join. Let me know if that is required.

Reply:
Impala uses HMS API for transaction management. They rely on a TxnHandler.lock() method to handle concurrent txns.

Reply:
I changed it to split & convert partition key to lower key.